.github/workflows/cmake.yml

```diff
@@ -0,0 +1,53 @@
+name: CMake
+
+on: [push, pull_request]
+
+jobs:
+
+  build:
+    strategy:
+      matrix:
+        BUILD_TYPE: [Release, Debug]
+      
+    # The CMake configure and build commands are platform agnostic and should work equally
+    # well on Windows or Mac.  You can convert this to a matrix build if you need
+    # cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-latest
```

Comment:
Later we should use our containers here:
```
 distro: ['opensuse', 'ubuntu_rolling', 'fedora', 'ubuntu', 'ubuntu_mpich', 'fedora_mpich']
container: espresso/buildenv:${{ matrix.distro }}
```

Reply:
If we want to maintain the containers, yes. Is there much difference between the different OSes? Or do you want to test if `espressopp` can be compiled with the packaged libraries?

Reply:
All distro have different version of all kind of packages and compilers hence it is good to test all of them....

Reply:
I am afraid custom Docker images are only available either by using self-hosted runners or running docker inside the VM. https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
I do not really like both solutions.

Reply:
Different compilers and OSes will be added in a separate MR.